### PR TITLE
requests: place a request for a patron by a librarian

### PR DESCRIPTION
**Pull request by BadrAly**
* Removes duplicate fixtures function.
* Adds sort on pickup_name location

Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>
Co-Authored-by: Aly Badr <aly.badr@rero.ch>

## Why are you opening this PR?

https://tree.taiga.io/project/rero21-reroils/us/657?milestone=257446

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

File: rero_ils/modules/items/utils.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+def jsonify_response(response=False, reason=None):
+    """Jsonify api response."""
+    return jsonify({
+        'can_request': response,
+        'reason': reason
+    })
+
+
+def is_librarian_can_request_item_for_patron(
```
Comment (jma) [inline rero_ils/modules/items/utils.py]: if it is not used at other place I would put this code in the view.

File: rero_ils/modules/items/utils.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        item_pid, library_pid, patron_barcode):
+    """Check if a librarian can request an item for a patron.
+
+    required_parameters: item_pid, library_pid, patron_barcode
+    """
+    item = Item.get_record_by_pid(item_pid)
+    if not item:
+        return jsonify_response(reason=_('Item not found.'))
```
Comment (jma) [inline rero_ils/modules/items/utils.py]: Translation does not makes sense here as it is exposed to the api.

File: rero_ils/modules/items/utils.py (hunk 1 above)
Comment (BadrAly) [reply rero_ils/modules/items/utils.py]: done

File: rero_ils/modules/items/utils.py (hunk 2 above)
Comment (BadrAly) [reply rero_ils/modules/items/utils.py]: done

File: rero_ils/modules/items/api_views.py [hunk 3]
```diff
@@ -323,3 +324,58 @@ def item_availability(item_pid):
...
+    '/<item_pid>/can_request/<library_pid>/<patron_barcode>', methods=['GET'])
+@check_authentication
+@jsonify_error
+def can_request(item_pid, library_pid, patron_barcode):
+    """HTTP request to check if a librarian can request an item for a patron.
+
+    required_parameters: item_pid, library_pid, patron_barcode
+    """
+    return is_librarian_can_request_item_for_patron(
+        item_pid, library_pid, patron_barcode)
```
Comment (zannkukai) [inline rero_ils/modules/items/api_views.py]: Why don't using immediately `is_librarian_can_request_item_for_patron` or insert the code of this method into ?

**Pull request merged**